src/Nethermind/Nethermind.Merge.Plugin/EngineRpcModule.Paris.cs

```diff
@@ -39,10 +39,13 @@ public async Task<ResultWrapper<PayloadStatusV1>> engine_newPayloadV1(ExecutionP
 
     private async Task<ResultWrapper<ForkchoiceUpdatedV1Result>> ForkchoiceUpdated(ForkchoiceStateV1 forkchoiceState, PayloadAttributes? payloadAttributes, int version)
     {
-        if (payloadAttributes?.Validate(_specProvider, version, out string? error) == false)
+        string? error = null;
+        switch (payloadAttributes?.Validate(_specProvider, version, out error))
         {
-            if (_logger.IsWarn) _logger.Warn(error);
-            return ResultWrapper<ForkchoiceUpdatedV1Result>.Fail(error, version >= EngineApiVersions.Cancun ? ErrorCodes.UnsupportedFork : ErrorCodes.InvalidParams);
+            case PayloadAttributesValidationResult.InvalidParams:
+                return ResultWrapper<ForkchoiceUpdatedV1Result>.Fail(error!, ErrorCodes.InvalidParams);
+            case PayloadAttributesValidationResult.UnsupportedFork:
+                return ResultWrapper<ForkchoiceUpdatedV1Result>.Fail(error!, ErrorCodes.UnsupportedFork);
```

Comment:
Why we need to map same labels to same labels? Do we need `PayloadAttributesValidationResult` at all?

Reply:
This one is harder to describe, we need it for refactoring.
We have the similar enum `Nethermind.Merge.Plugin.Data.ValidationResult` for `ExecutionPayload` validation that also includes value that means `ErrorCode=None, Result=Invalid`. So I decided to make a separate `enum` that will merge with this one and will be different from `ErrorCodes`.

The current solution for fcu seems more correct that ones we have for get/new payload. And I expected to find more time later to refactore 3 of them to follow this one approach.

Reply:
Ah it maps to int's got it, though I would just `out` the error code from `Validate` method and make it bool and drop the enum.